test/jdk/java/lang/module/ModuleDescriptorHashCodeTest.java

```diff
@@ -63,6 +62,105 @@ public void testBootModuleDescriptor() throws Exception {
         }
     }
 
+    /**
+     * Verifies that two "equal" module descriptors which only differ in the order of
+     * {@link ModuleDescriptor.Opens.Modifier opens modifiers}, that were used to construct the
+     * descriptors, have the same hashcode.
+     */
+    @Test
+    public void testOpensModifiersOrdering() throws Exception {
+        // important to use Set.of() (i.e. backed by immutable set) to reproduce the issue
+        final Set<ModuleDescriptor.Opens.Modifier> mods1 = Set.of(
+                ModuleDescriptor.Opens.Modifier.SYNTHETIC, ModuleDescriptor.Opens.Modifier.MANDATED);
+        final ModuleDescriptor desc1 = createModuleDescriptor(mods1, null, null);
+
+        // create the same module descriptor again and this time just change the order of the
+        // "opens" modifiers' Set.
+
+        // important to use Set.of() (i.e. backed by immutable set) to reproduce the issue
+        final Set<ModuleDescriptor.Opens.Modifier> mods2 = Set.of(
+                ModuleDescriptor.Opens.Modifier.MANDATED, ModuleDescriptor.Opens.Modifier.SYNTHETIC);
+        final ModuleDescriptor desc2 = createModuleDescriptor(mods2, null, null);
```

Comment:
If you static important Requires, Exports and Opens then it should mean you can fit these onto a single line and might be easier to read, e.g.
```
var mods2 = Set.of(Opens.Modifier.MANDATED, Opens.Modifier.SYNTHETIC);
```